File: backend/crates/akashic-extraction/tests/fixtures/dart/01-class.dart
class Counter {
  int _count;

  Counter(this._count);

  int increment() {
    _count = _count + 1;
    return _count;
  }

  int incrementTwice() {
    increment();
    increment();
    return _count;
  }

  void _reset() {
    _count = 0;
  }
}

File: backend/crates/akashic-extraction/tests/fixtures/scala/01-class.scala
class Counter(start: Int) {
  private var count: Int = start

  def increment(): Int = {
    count = count + 1
    count
  }

  def incrementTwice(): Int = {
    increment()
    increment()
  }

  protected def reset(): Unit = {
    count = 0
  }
}
